parsers/cxx/cxx_token_chain.c

```diff
@@ -1024,6 +1040,8 @@ CXXToken * cxxTokenChainExtractRange(
 		vStringPut (pRet->pszWord, ' ');
 	pRet->bFollowedBySpace = pToken->bFollowedBySpace;
 
+	cxxSideChainAppend(pToken, pRet);
```

Comment:
This should be done in upper layers.

Reply:
See cxxParserEmitFunctionTags() in cxx_parser_function.c.
```
	CXXToken * pIdentifier = cxxTokenChainExtractRange(
			pInfo->pIdentifierStart,
			pInfo->pIdentifierEnd,
			// proper spacing has been already ensured
			// by cxxParserLookForFunctionSignature()
			0
		);

	cxxTokenChainDestroyRange(pInfo->pIdentifierChain,pInfo->pIdentifierStart,pInfo->pIdentifierEnd);
```
Before destroying, call `cxxTokenChainExtractRange (pInfo->pIdentifierStart,pInfo->pIdentifierEnd, pIdentifier);`

The `CXX_DEBUG_ASSERT` in the next line should be called before 
`cxxTokenChainExtractRange()`.